batchglm/train/numpy/glm_nb/estimator.py

```diff
@@ -21,19 +20,17 @@ class Estimator(EstimatorGlm):
 
     def __init__(
         self,
-        input_data: InputDataGLM,
-        init_location: Union[np.ndarray, str] = "AUTO",
-        init_scale: Union[np.ndarray, str] = "AUTO",
-        batch_size: Optional[Union[Tuple[int, int], int]] = None,
+        init_location: str = "AUTO",
+        init_scale: str = "AUTO",
+        # batch_size: Optional[Union[Tuple[int, int], int]] = None,
         quick_scale: bool = False,
+        model=None,
         dtype="float64",
         **kwargs
     ):
         """
         Performs initialisation and creates a new estimator.
 
-        :param input_data: InputDataGLM
-            The input data
         :param init_location: (Optional)
```

Comment:
Would it make sense to make this an enum internally?  The if-else would at least look nicer.

Reply:
What should be an enum? and what if-else are you referring to here?

Reply:
ah ok this is about the utils init_par I guess...

Reply:
Yes, but we could type check the enum, I think